### ISSUE TITLE
Unable to deploy RC5 app to Google Cloud App Engine

### ISSUE THREAD

**Issue by antonyRoberts** [angular/angular-cli#1743]
> Please provide us with the following information:
> ---------------------------------------------------------------

1. OS?  Windows 10

2. Versions?
angular-cli: local (v1.0.0-beta.11-webpack.2, branch: master)
node: 6.3.1

3. Brand new App created with the CLI using master. Add app.yaml file and change package.json to include reference to node engine and preinstall angular-cli on the server.  (This setup works fine for an RC4 app using the CLI). When deploying to google cloud with `gcloud app deploy --verbosity debug` it fails with:
 `Updating service [default]...failed.
DEBUG: (gcloud.app.deploy) Error Response: [13] Timed out when starting VMs.  It's possible that the application code is unhealthy.  (0/2 ready, 2 still deploying).
Traceback (most recent call last):`

4. This is the output from the debug result.
[gcloudDeployVerboseDebug.txt](https://github.com/angular/angular-cli/files/425101/gcloudDeployVerboseDebug.txt)

5. Mention any other details that might be useful.
I'll be on vacation for the next few days and not able to respond.
> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

**Comment by kylecordes** [comment]
The output of a CLI build is just a directory with some static files in it. It has nothing whatever to do with deploying to any particular hosting. It sounds like you're trying to actually execute the CLI on Google app engine - a very unusual thing to do, and almost certainly not useful. Instead, set things up instead so that you run the CLI build command and then you set things up with your hosting mechanism (Google app engine in your case) to upload the resulting files.

**Comment by antonyRoberts** [comment]
You are correct that is what I'm trying to do and I agree it does not sound like the right way to do it at all.  It's just the only way I was able to make it deploy to the appEngine when using RC4 there is no documentation concerning this at the moment, the most helpful thing I saw was this little blog post. http://jaykhimani.blogspot.com/2016/08/deploying-angular-2-typescript-app-with.html

Are you suggesting then not to use gcloud's deploy commands and instead set up a static VM and just upload the build files?

**Comment by kylecordes** [comment]
It is probably best to move the conversation elsewhere, as this is more of a support and assistance request rather than Angular issue. (running away before @pkozlowski-opensource chases me away ;-) )

**Comment by filipesilva** [comment]
I'm sorry but there isn't much help I can give on this. Like @kylecordes said, the output of the CLI building the app is just a bunch of html and css files, so you should be able to deploy it as a static website. 

There isn't any GAE specific that I am aware of really. Try to debug it by making a very simple index.html that you want to serve, and making sure you're able to serve that. If that doesn't work it's a good indication that something is amiss. Once that works, you should be able to just serve whatever is in `dist/` after `ng build`.

**Comment by antonyRoberts** [comment]
@filipesilva Thanks for the fresh set of eyes, I appreciate it.  You are correct in your statement that using the build files is the way to go. The issue I had in doing that was with google Cloud's issue with not allowing https for CNAME redirects when using their Cloud Storage which is the suggestion of hosting a static site by google on the cloud platform. https://cloud.google.com/storage/docs/hosting-static-website this is a troublesome suggestion.  when I tried to use gcloud app deploy with the dist folder it would error out with a need for the app.yaml file which I'd butchered before following a blogs post to get GAE working with the CLI on RC4.  Eventually I created a separate folder for the app to be distributed from with a simpler app.yaml file which does nothing more than point to index.html in a static folder containing the distribution files.  The documentation to do so is lacking, I might do a little write up in the next few days to help others who are bound to run into this problem.
